### ISSUE TITLE
NullReferenceException using XPath searching for //p

### ISSUE THREAD

**Issue by sofdwere** [dnGrep/dnGrep#919]
dnGrep v4.0.69.0   x64    on Win10

I have reduced the problematic file to this simple example:

```
<?xml version="1.0" encoding="utf-8" ?>
<html>
<head>
</head>
<body>
<ul>
<li>
Text1<br/>Text2<p>Text3</p>
</li>
</ul>
</body></html>
```

Searching for //p using XPath throws Exception, why?

```
<logevent time="2023-08-14 15:05:45.5348" level="Error" logger="dnGREP.WPF.MainViewModel">
  <message>Failure in search progress changed</message>
  <exception>System.NullReferenceException: Object reference not set to an instance of an object.
   at dnGREP.WPF.FormattedGrepResult.&lt;&gt;c.&lt;SetLabel&gt;b__37_0(GrepMatch r) in C:\projects\dngrep\dnGrep.WPF\ViewModels\GrepSearchResultsViewModel.cs:line 520
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.DistinctIterator`1.GetCount(Boolean onlyIfCheap)
   at dnGREP.WPF.FormattedGrepResult.SetLabel() in C:\projects\dngrep\dnGrep.WPF\ViewModels\GrepSearchResultsViewModel.cs:line 531
   at dnGREP.WPF.FormattedGrepResult..ctor(GrepSearchResult result, String folderPath) in C:\projects\dngrep\dnGrep.WPF\ViewModels\GrepSearchResultsViewModel.cs:line 493
   at dnGREP.WPF.GrepSearchResultsViewModel.AddRange(List`1 list) in C:\projects\dngrep\dnGrep.WPF\ViewModels\GrepSearchResultsViewModel.cs:line 201
   at dnGREP.WPF.MainViewModel.SearchProgressChanged(Object sender, ProgressChangedEventArgs e) in C:\projects\dngrep\dnGrep.WPF\ViewModels\MainViewModel.cs:line 1203</exception>
</logevent>
```

**Comment by doug24** [comment]
The reason this is failing is the text nodes, `Text1` and `Text2` - the existing dnGrep code doesn't handle them properly.  I fixed it by just adding one new case to a switch statement and that didn't break any unit tests, which is a good thing.  I see why the code was written this way, but it seems very fragile - I wouldn't be surprised if there are other xml constructs that aren't being handled correctly either.  I think I'll add this fix for now, but this is an area that could use more attention or a rewrite.

**Comment by doug24** [comment]
Fixed in v4.0.115
